homeassistant/components/lutron_caseta/device_trigger.py

```diff
@@ -484,3 +490,14 @@ def get_button_device_by_dr_id(hass: HomeAssistant, device_id: str):
             return device
 
     return None
+
+
+def is_lutron_caseta_device(hass: HomeAssistant, device_id: str) -> bool:
+    """Check if HomeAssistant device is port of this domain."""
+    device_registry = dr.async_get(hass)
+    is_lutron_caseta = False
+    if device_entry := device_registry.async_get(device_id):
+        is_lutron_caseta = any(
+            identifier[0] == DOMAIN for identifier in device_entry.identifiers
+        )
+    return is_lutron_caseta
```

Comment:
We shouldn't make the code more complex to satisfy the tests, instead we should have an explicit test for the branching.